Source/WTF/wtf/text/Base64.cpp

```diff
@@ -267,4 +268,18 @@ std::optional<Vector<uint8_t>> base64Decode(StringView input, Base64DecodeMode m
     return base64DecodeInternal(std::span(input.characters16(), length), mode);
 }
 
+std::optional<String> base64DecodeToString(StringView input, Base64DecodeMode mode)\
+{
+    auto toString = [&] (auto optionalBuffer) -> std::optional<String> {
```

Comment:
I think this PR really shows that we need to introduce a `NonNullableString` in the codebase at some point or at least a `Markable<String>`.

In the meantime, I guess we should use the `isNull()` function instead of using an `std::optional<String>`.

Reply:
Ah, you're right. I was thinking for some reason that `String::adopt` would return the nullString if given an empty vector but that's not the case. Will change the API.